src/core/src/op/paged_attention.cpp

```diff
@@ -23,165 +105,25 @@ void PagedAttentionExtension::validate_and_infer_types() {
-    NODE_VALIDATION_CHECK(this,
-                          get_input_element_type(9).is_dynamic() || get_input_element_type(9).is_real(),
-                          "Element type of `scale` input should be a floating type, but it is ",
-                          get_input_element_type(9),
-                          ".");
-    NODE_VALIDATION_CHECK(
-        this,
-        get_input_partial_shape(10).rank().is_dynamic() || get_input_partial_shape(10).rank().get_length() == 0,
-        "Input `sliding_window` should be a scalar but it has rank ",
-        get_input_partial_shape(10).rank().get_length(),
-        ".");
-    NODE_VALIDATION_CHECK(this,
-                          get_input_element_type(10).is_dynamic() || get_input_element_type(10) == element::i32,
-                          "Element type of `sliding_window` input should be i32, but it is ",
-                          get_input_element_type(10),
-                          ".");
-
-    NODE_VALIDATION_CHECK(
-        this,
-        get_input_partial_shape(11).rank().is_dynamic() || get_input_partial_shape(11).rank().get_length() == 1,
-        "Rank of `alibi_slopes` input should be 1, but it is ",
-        get_input_partial_shape(11).rank().get_length(),
-        ".");
-    NODE_VALIDATION_CHECK(this,
-                          get_input_element_type(11).is_dynamic() || get_input_element_type(11).is_real(),
-                          "Element type of `alibi_slopes` input should be a floating type, but it is ",
-                          get_input_element_type(11),
-                          ".");
-    NODE_VALIDATION_CHECK(
-        this,
-        get_input_partial_shape(12).rank().is_dynamic() || get_input_partial_shape(12).rank().get_length() == 0,
-        "Input `max_context_len` should be a scalar but it has rank ",
-        get_input_partial_shape(12).rank().get_length(),
-        ".");
-    NODE_VALIDATION_CHECK(this,
-                          get_input_element_type(12).is_dynamic() || get_input_element_type(12) == element::i32,
-                          "Element type of `max_context_len` input should be i32, but it is ",
-                          get_input_element_type(12),
-                          ".");
+    // format: Node*, input_idx, name, {rank_list}, {type_list}
+    input_check(this, 0, "query", {2}, {});
+    input_check(this, 1, "key", {2}, {});
+    input_check(this, 2, "value", {2}, {});
+    input_check(this, 3, "key_cache", {2, 3, 4, 5}, {});
+    input_check(this, 4, "value_cache", {2, 3, 4, 5}, {});
+    input_check(this, 5, "past_lens", {1}, {element::i32});
+    input_check(this, 6, "subsequence_begins", {1}, {element::i32});
+    input_check(this, 7, "block_indices", {1}, {element::i32});
+    input_check(this, 8, "block_indices_begins", {1}, {element::i32});
+    input_check(this, 9, "scale", {0}, get_real_types());
+    input_check(this, 10, "sliding_window", {0}, {element::i32});
+    input_check(this, 11, "alibi_slopes", {1}, get_real_types());
```

Comment:
Port names can be defined as local constexpr array and port index can be used to access it.

Reply:
the current approach is more human readable, we see the mapping, idx, ranks, types as single line